lib/koans/19_streams.ex

```diff
@@ -0,0 +1,27 @@
+defmodule Streams do
+  use Koans
+
+  @intro "Streams"
+
+  koan "Streams are lazier than Enum, they only return a struct" do
+    assert Stream.map([1, 2, 3], &(&1 + 1)).__struct__ == ___
+  end
+
+  koan "You have to ask it explicitly to do the operation" do
+    updated_list =
+      [1, 2, 3]
+      |> Stream.map(&(&1 + 1))
```

Comment:
We're missing something important in these examples, an illustration that streams avoid iterating over a collection multiple times. That might be hard to illustrate, do you have any ideas? At the very least, we should probably have an example that shows piping multiple Stream calls together to make the point that each one does not represent an iteration through the collection, but rather a "stacking" of the streamed behavior in the pipeline.